src/canopy_alloc.F90

```diff
@@ -1,380 +1,658 @@
 
-SUBROUTINE canopy_alloc
+!> \file canopy_alloc.F90
+!! \brief Memory allocation subroutine for canopy model arrays
+!! \author P.C. Campbell
+!! \date 03 Oct 2022
+!! \version 1.0
 
-!-------------------------------------------------------------------------------
-! Name:     Allocate Arrays for Canopy Inputs and Outputs
-! Purpose:  Allocate arrays for Canopy Inputs and Outputs
-! Revised:  03 Oct 2022  Original version.  (P.C. Campbell)
-!-------------------------------------------------------------------------------
+!> \defgroup MemoryManagement Memory Management
+!! \brief Routines    !> \defgroup Daily3DAverages Daily Average 3D Arrays
+    !! \brief 3D versions of daily average arrays
+    !! \ingroup MemoryManagement
+    !! \{
+    !> \brief 24-hour average sunlit leaf temperature 3D (K)locating and deallocating memory for canopy model arrays
```

Comment:
This is the part it wants to dedent. Seems like there is some junk in this block anyway.